lib/remote_console/web_mks.rb

```diff
@@ -0,0 +1,41 @@
+module RemoteConsole
+  class WebMKS < SSLSocket
+    attr_accessor :url
+
+    def initialize(record, socket)
+      super(record, socket)
+      @url = URI::Generic.build(:scheme => 'wss',
+                                :host   => @record.host_name,
+                                :port   => @record.port,
+                                :path   => path).to_s
+      @driver = WebSocket::Driver.client(self, :protocols => [protocol])
+      @driver.on(:close) { socket.close unless socket.closed? }
+      @driver.start
+    end
+
+    def fetch(length)
+      @driver.on(:message) { |msg| yield(msg.data) } if @driver.listeners(:message).length.zero?
```

Comment:
this was a little confusing for me, so I'm including for others

this is setting a callback if there are not callbacks already set. This is the first time `fetch` is called

Reply:
Well, the websocket-driver operates with callbacks, so there's no prettier solution. The actual fetching is done by the lines below, this is just handing over the parsed message to a block. As the callback cannot be unregistered, I'm setting it only once and I rely on the context. It's ugly I know, but it works :confused:

Reply:
It's not about customization, in the moment of creating the adapter there's no knowledge about the "other end".